### PR TITLE
{lib}[dummy] NCCL 2.1.4 w/ CUDA 9.0.176

### PR DESCRIPTION
**Pull request by damianam**

### CODE REVIEW

**Comment by damianam** [comment]
Test report

https://gist.github.com/damianam/36a0a95529122d060a7ab3667b2a031f

File: easybuild/easyconfigs/n/NCCL/NCCL-2.1.4-CUDA-9.0.176.eb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+cuda_version = '9.0.176'
+versionsuffix = '-CUDA-%s' % cuda_version
+
+homepage = 'https://developer.nvidia.com/nccl'
+description = """The NVIDIA Collective Communications Library (NCCL) implements multi-GPU and multi-node collective
+communication primitives that are performance optimized for NVIDIA GPUs."""
+
+toolchain = {'name': 'dummy', 'version': ''}
+
+sources = ['%%(namelower)s_%%(version)s-1+cuda%s_x86_64.txz' % '.'.join(cuda_version.split('.')[0:2])]
```
Comment (boegel) [inline easybuild/easyconfigs/n/NCCL/NCCL-2.1.4-CUDA-9.0.176.eb]: @damianam Please include a comment to clarify where to download from (https://developer.nvidia.com/nccl/nccl-download).

Also: include SHA256 checksum

File: easybuild/easyconfigs/n/NCCL/NCCL-2.1.4-CUDA-9.0.176.eb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+dependencies = [
+    ('CUDA', cuda_version)
+]
+
+skipsteps = ['build']
+
+files_to_copy = ['lib', 'include']
+
+sanity_check_paths = {
+    'files': ['lib/libnccl.so', 'lib/libnccl_static.a', 'include/nccl.h'],
```
Comment (boegel) [inline easybuild/easyconfigs/n/NCCL/NCCL-2.1.4-CUDA-9.0.176.eb]: @damianam Please use `'lib/libnccl.%s' % SHLIB_EXT` rather than `lib/libnccl.so`

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
gligar02.gligar.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/55f69eaf70f13a94d56231a620337bf0 for a full test report.

**Comment by damianam** [comment]
@boegel took a bit, but the changes came ;-)

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2107.delcatty.os - Linux centos linux 7.5.1804, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/83811d71b763a6313aadc3810897761d for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node3102.skitty.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 2.7.5
See https://gist.github.com/d997b92a96b78542cbaa6f7f4ece8557 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2570.golett.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/f699fac31e383fadde138cc5018e25b4 for a full test report.

**Comment by boegel** [comment]
Going in, thanks @damianam!

**Pull request merged**